### ISSUE TITLE
Cannot read property 'replace' of undefined

### ISSUE THREAD

**Issue by briannc29** [knsv/mermaid#239]
I try the sample code on doc and I got the error : "cannot read property 'replace' of undefined". The error at the line mermaidAPI.render(). How to fix it?

<!DOCTYPE html>
````<html>
<head>
    <meta charset="UTF-8">
	<script src="jquery-1.11.3.min.js"></script>
    <link rel="stylesheet" href="mermaid.css" />
    <script src="mermaidAPI.js"></script>
	
	<script>
		mermaidAPI.initialize({
			startOnLoad:false
		});
        // Example of using the API
		$(function(){
			var graphDefinition = 'graph TB\na-->b';
			var graph = mermaidAPI.render(graphDefinition);
			$("#graphDiv").html(graph);
		});
		
	</script>
</head>
<body>
	
    <div class="mermaid" id="graphDiv">
	
    </div>

</body>
</html>
````

**Comment by knsv** [comment]
Small change in the render call:
```
    <script>
        mermaidAPI.initialize({
            startOnLoad:false
        });
        $(function(){
        // Example of using the API
            var element = document.querySelector("#graphDiv");

            var insertSvg = function(svgCode, bindFunctions){
                element.innerHTML = svgCode;
            };

            var graphDefinition = 'graph TB\na-->b';
            var graph = mermaidAPI.render('graphDiv', graphDefinition, insertSvg);
        });
    </script>
````

I noticed the docs was not updated. Will update them.

**Comment by tylerlong** [comment]
Hi @knsv ,

I thought mermaidAPI.js was generating HTML on server side (aka, node.js). If mermaidAPI.js still depends on DOM (document, innerHTML...etc), what's the advantage of it over mermaid.js ? 

With mermaid.js, I can do the following which is even simpler and more straightforward:

```javascript
$('#graphDiv').text('graph TB\na-->b');
$('#graphDiv').addClass('mermaid');
mermaid.init();
```

**Comment by knsv** [comment]
The origins of the API was to separate the site integrations from the rendering of diagrams. The idea was to make complex integrations easier and somewhat cleaner and at the same time not inpose more complexity in the original style of integration which was starting to happen.

All that said, I have no plans of removing the original integration as in your first example. The two integration style will live on side by side. mermaid.init in tun uses the API for generation the graph so there are no code duplication due to this.

**Comment by tylerlong** [comment]
Understood. I still have two questions:

1. why do you remove the following API:

```
var graphDefinition = 'graph TB\na-->b';
var graph = mermaidAPI.render(graphDefinition);
$("#graphDiv").html(graph);
```

2.  `var graph = mermaidAPI.render('graphDiv', graphDefinition, insertSvg);` graph is undefined which means `mermaidAPI.render` returns nothing. Is it by design?

**Comment by knsv** [comment]
I see what you mean now.

I had some issues with asynchronous rendering it was not always ready rendered straight away. This had to do with subgraphs and dagre-d3. That is no longer the case thanks to a new version of dagre-d3.

Perhaps a straight forward way would be to return as you propose when no callback is supplied. Easier interface and still backwards compatible.

**Comment by knsv** [comment]
Closing issue as it's resolution is included in release 0.5.6. Let me know if you do not agree.
